src/streaming.ts

```diff
@@ -46,46 +52,116 @@ export class StreamProcessor {
     this.eventBus = eventBus;
     this.repository = repository;
     this.log = this.options.logger;
+
+    const minDelay = 5000;
+    const maxDelay = 10000;
+    this.retryAttempt = 1;
+    this.retryDelayMs = Math.floor(
+      Math.random() * (maxDelay - minDelay + 1) + minDelay,
+    );
   }
 
   start(): void {
-    this.log.info('Starting StreamProcessor');
-    const eventSource = new EventSource(
-      `${this.options.baseUrl}/stream?cluster=${this.cluster}`,
-      {
-        headers: {
-          Authorization: `Bearer ${this.jwtToken}`,
-          'API-Key': this.apiKey,
-        },
+    this.log.info('Starting new StreamProcessor');
+
+    const url = `${this.options.baseUrl}/stream?cluster=${this.cluster}`;
+
+    const options = {
+      headers: {
+        'Cache-Control': 'no-cache',
+        Accept: 'text/event-stream',
+        Authorization: `Bearer ${this.jwtToken}`,
+        'API-Key': this.apiKey,
       },
-    );
+    };
 
-    eventSource.onopen = (event: MessageEvent) => {
-      this.log.debug('Stream connected', event);
+    const connected = () => {
```

Comment:
It's usually good practice to prefix handlers with `on` to set them apart. Or use the suffix `Handler`. Or both.
```suggestion
    const onConnectedHandler = () => {
```